tensilelite/Tensile/TensileCreateLibrary.py

```diff
@@ -244,93 +230,83 @@ def buildSourceCodeObjectFile(CxxCompiler, outputPath, kernelFile):
-              baseVariant = base+"-"+variant if variant else base
-              if arch in baseVariant:
-                outfile = os.path.join(buildPath, baseVariant+".hsaco")
-              else:
-                outfile = None
-            else:
-              outfile = os.path.join(buildPath, "{0}-000-{1}.hsaco".format(soFilename, arch))
-
-            #Compilation
-            if outfile:
-              coFilenames.append(os.path.split(outfile)[1])
-              #bundlerArgs = [globalParameters["ClangOffloadBundlerPath"], "-type=o", "-targets=%s" % target, "-inputs=%s" % infile, "-outputs=%s" % outfile, "-unbundle"]
-              bundlerArgs = [globalParameters["ClangOffloadBundlerPath"], "-type=o", "-targets=%s" % target,
-                           "%s=%s" % (inflag, infile), "%s=%s" % (outflag, outfile), "-unbundle"]
-              if globalParameters["PrintCodeCommands"]:
-                print(' '.join(bundlerArgs))
-              subprocess.check_call(bundlerArgs)
-
-      except subprocess.CalledProcessError:
-        for i in range(len(archs)):
-          outfile = os.path.join(buildPath, "{0}-000-{1}.hsaco".format(soFilename, archs[i]))
-          coFilenames.append(os.path.split(outfile)[1])
-          #bundlerArgs = [globalParameters["ClangOffloadBundlerPath"], "-type=o", "-targets=hip-amdgcn-amd-amdhsa--%s" % cmdlineArchs[i], "-inputs=%s" % infile, "-outputs=%s" % outfile, "-unbundle"]
-          bundlerArgs = [globalParameters["ClangOffloadBundlerPath"], "-type=o", "-targets=hip-amdgcn-amd-amdhsa--%s" % cmdlineArchs[i],
-                         "%s=%s" % (inflag, infile), "%s=%s" % (outflag, outfile), "-unbundle"]
-          if globalParameters["PrintCodeCommands"]:
-            print(' '.join(bundlerArgs))
-          subprocess.check_call(bundlerArgs)
+    if os.name == "nt":
+      hipFlags += ['-fms-extensions', '-fms-compatibility', '-fPIC', '-Wno-deprecated-declarations']
+      compileArgs = launcher + [which(CxxCompiler)] + hipFlags + archFlags + [kernelFile, '-c', '-o', os.path.join(buildPath, objectFilename)]
     else:
-      raise RuntimeError("Unknown compiler {}".format(CxxCompiler))
+      compileArgs = launcher + [which(cxxCompiler)] + hipFlags + archFlags + [kernelFile, '-c', '-o', os.path.join(buildPath, objectFilename)]
+
+    if globalParameters["PrintCodeCommands"]:
+      print(cxxCompiler + ':' + ' '.join(compileArgs))
+    subprocess.check_call(compileArgs)
+
+    # If we aren't using hipcc what happens?
+    # get hipcc version due to compatiblity reasons
+    hipccver = globalParameters['HipClangVersion'].split(".")
+    hipccMaj = int(hipccver[0])
+    hipccMin = int(hipccver[1])
+    # for hipclang 5.2 and above, clang offload bundler changes the way input/output files are specified
+    inflag = "-inputs"
+    outflag = "-outputs"
+    if (hipccMaj == 5 and hipccMin >= 2) or hipccMaj >= 6:
+      inflag = "-input"
+      outflag = "-output"
```

Comment:
seems like we could do most of this once when we validate our toolchain

Reply:
Great idea, this code will also be affected by #1374, after which the toolchain consolidation work can be concluded in a final PR.